api/api/test/test_authentication.py

```diff
@@ -72,44 +72,47 @@ def test_check_user(mock_raise_if_exc, mock_submit, mock_distribute_function, mo
     mock_raise_if_exc.assert_called_once()
 
 
-@patch('api.authentication.token_urlsafe', return_value='test_token')
+@patch('api.authentication.change_keypair', return_value=('-----BEGIN PRIVATE KEY-----',
+                                                          '-----BEGIN PUBLIC KEY-----'))
 @patch('os.chmod')
-@patch('api.authentication.chown')
+@patch('os.chown')
 @patch('builtins.open')
-def test_generate_secret(mock_open, mock_chown, mock_chmod, mock_token):
-    """Check if result's length and type are as expected and mocked function are called with correct params."""
-    result = authentication.generate_secret()
-    assert isinstance(result, str)
-    assert result == 'test_token'
-
-    calls = [call(authentication._secret_file_path, mode='x')]
-    mock_open.has_calls(calls)
-    mock_chown.assert_called_once_with(authentication._secret_file_path, authentication.wazuh_uid(),
-                                       authentication.wazuh_gid())
-    mock_chmod.assert_called_once_with(authentication._secret_file_path, 0o640)
+def test_generate_keypair(mock_open, mock_chown, mock_chmod, mock_change_keypair):
+    """Verify correct params when calling open method inside generate_keypair"""
+    result = authentication.generate_keypair()
+    assert isinstance(result[0], str)
+    assert isinstance(result[1], str)
+    assert result == ('-----BEGIN PRIVATE KEY-----',
+                      '-----BEGIN PUBLIC KEY-----')
+    
+    calls = [call(authentication._private_key_path, authentication.wazuh_uid(), authentication.wazuh_gid()),
+             call(authentication._public_key_path, authentication.wazuh_uid(), authentication.wazuh_gid())]
+    mock_chown.assert_has_calls(calls)
+    calls = [call(authentication._private_key_path, 0o640),
+             call(authentication._public_key_path, 0o640)]
+    mock_chmod.assert_has_calls(calls)
 
     with patch('os.path.exists', return_value=True):
-        authentication.generate_secret()
-
-        calls.append(call(authentication._secret_file_path, mode='r'))
+        calls = [call(authentication._private_key_path, mode='rb'), call(authentication._public_key_path, mode='rb')]
         mock_open.has_calls(calls)
```

Comment:
There seems to be something wrong. This test should not pass as the `open` method doesn't have these calls (we need another call to `authentication.generate_keypair()` (inside the context of `os.path.exists` with `return_value=True`).

In case we had the `authentication.generate_keypair()`, the test should still fail as the calls are not those ones.